game/models.py

```
@@ -136,4 +162,30 @@ class Meta:
+    )
+
+    def save(self, *args, **kwargs):
+        if not self.pk:
+            count = FeaturedBadge.objects.filter(
+                user=self.user
+            ).count()
+
+            if count >= 3:
+                raise ValueError(
+                    "Users can only feature up to 3 badges"
+                )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)